--- a/cmake/FindJsonCPP.cmake	2025-04-26 19:30:45.485615655 +0800
+++ b/cmake/FindJsonCPP.cmake	2025-04-26 19:28:55.238152126 +0800
@@ -58,26 +58,20 @@
 
 else()
 	
-	set(_OPTIONS "CONFIG")
-	if(JsonCPP_FIND_REQUIRED)
-		list(APPEND _OPTIONS "REQUIRED")
-	endif()  
-	if(JsonCPP_FIND_QUIETLY)
-		list(APPEND _OPTIONS "QUIET")
-	endif()
-	if(JsonCPP_FIND_VERSION)
-		list(PREPEND _OPTIONS "${jsoncpp_FIND_VERSION}")
-	endif()
-	if(JsonCPP_FIND_EXACT)
-		list(APPEND _OPTIONS "EXACT")
-	endif()
 
-	find_package(jsoncpp ${_OPTIONS})
-
-	if(TARGET jsoncpp_lib)
-		set(JsonCPP_TARGET jsoncpp_lib)
-	else()
-		set(JsonCPP_TARGET jsoncpp_static)
-	endif()
+	find_package(PkgConfig QUIET REQUIRED)
+	pkg_check_modules(JSONCPP REQUIRED jsoncpp)
+ 
+	set(JsonCPP_INCLUDE_DIRS ${JSONCPP_INCLUDE_DIRS})
+	set(JsonCPP_LIBRARIES   ${JSONCPP_LIBRARIES})
+ 
+	include(FindPackageHandleStandardArgs)
+	find_package_handle_standard_args(JsonCPP
+		REQUIRED_VARS JsonCPP_INCLUDE_DIRS JsonCPP_LIBRARIES
+		VERSION_VAR JsonCPP_VERSION
+	)
+	mark_as_advanced(JsonCPP_INCLUDE_DIRS JsonCPP_LIBRARIES)
+ 
+	set(JsonCPP_TARGET jsoncpp CACHE STRING "Link target for JsonCPP")
 
 endif()

